youtube_dl/downloader/hls.py

```diff
@@ -28,9 +28,12 @@ def real_download(self, filename, info_dict):
             return False
         ffpp.check_version()
 
+        cookies = ''
+        for cookie in self.ydl.cookiejar:
+            cookies += '%s=%s; path=%s; domain=%s;\r\n' % (cookie.name, cookie.value, cookie.path, cookie.domain)
         args = [
             encodeArgument(opt)
-            for opt in (ffpp.executable, '-y', '-i', url, '-f', 'mp4', '-c', 'copy', '-bsf:a', 'aac_adtstoasc')]
+            for opt in (ffpp.executable, '-y', '-user-agent', info_dict['http_headers']['User-Agent'], '-cookies', cookies,'-i', url, '-f', 'mp4', '-c', 'copy', '-bsf:a', 'aac_adtstoasc')]
```

Comment:
i send only the cookies and user agent because they are the only information that i find some sites test for